metricbeat/module/elasticsearch/ccr/ccr.go

```diff
@@ -107,3 +114,52 @@ func (m *MetricSet) Fetch(r mb.ReporterV2) {
 		return
 	}
 }
+
+func (m *MetricSet) checkCCRAvailability(currentElasticsearchVersion string) (message string, err error) {
+	license, err := elasticsearch.GetLicense(m.HTTP, m.getServiceURI())
+	if err != nil {
+		return "", errors.Wrap(err, "error determining Elasticsearch license")
+	}
+
+	licenseType, err := license.GetValue("type")
+	if err != nil {
+		return "", errors.Wrap(err, "error determining Elasticsearch license type")
+	}
+
+	currentLicense, ok := licenseType.(string)
+	if !ok {
+		return "", fmt.Errorf("error determining Elasticsearch license type")
+	}
```

Comment:
When I was proposing to move the license checking to a common function I was thinking in reusing all this code needed to get the license name, but at the end we still need it here to generate the error code :confused:

The license checker could return the current license too, or we could have a getter for the license and pass it to `IsLicenseOf` so it doesn't need to be requested multiple times.

In any case I don't want to block this PR with this, we can think more about this if it has to be reused more in the future.

Reply:
Ugh, you're right, and that was my intent as well. Somehow I spaced and didn't refactor correctly :( Fixing now...

Reply:
As I'm attempting this refactoring (properly this time 😄) it seems that License is really asking to be a struct type with a method like `isOneOf`. So I'm going to go down this path now and see if we like it.

Reply:
@jsoriano I've done the refactoring for the license code in 812b90fb7. It ended up touching files beyond the scope of this PR, though, so if you'd like I can break out the license refactoring into its own PR to make reviewing easier.